src/core/project/qgsprojectdisplaysettings.h

```diff
@@ -90,6 +102,20 @@ class CORE_EXPORT QgsProjectDisplaySettings : public QObject
      */
     const QgsGeographicCoordinateNumericFormat *geographicCoordinateFormat() const;
 
+    /**
+     * Returns default coordinate format for the project.
+     * \see setCoordinateFormat()
+     * \since QGIS 3.28
+     */
+    Qgis::CoordinateFormat coordinateFormat() const { return mCoordinateFormat; }
+
+    /**
+     * Sets the default coordinate format for the project.
+     * \see coordinateFormat()
+     * \since QGIS 3.28
+     */
+    void setCoordinateFormat( Qgis::CoordinateFormat format );
```

Comment:
I'm 50/50 on this. It might be a good time to update the coordinate format to something more flexible -- ie. instead of a choice between Geographic (ie EPSG4326) and "map units", what if we instead had something like "ProjectCrs" / "SpecificCrs" and accompany this with a setCoordinateDisplayCrs( ... ) method. And then replace all our use of the old geographic value with setting the coordinate display crs to EPSG4326 explicitly. The rationale behind this would be to support non-earth situations, where having a fixed "geographic" === "epsg:4326" assumption is wrong. (Or say to support the use case where you have a project in 3857 but want to see coordinate values in a local CRS). Thoughts?